include/CL/sycl/parallelism/detail/parallelism.hpp

```diff
@@ -283,7 +283,7 @@ void parallel_for_workgroup(nd_range<Dimensions> r,
 template <std::size_t Dimensions, typename ParallelForFunctor>
 void parallel_for_workitem(const group<Dimensions> &g,
                            ParallelForFunctor f) {
-#if defined(_OPENMP) && !defined(TRISYCL_NO_BARRIER)
+#if defined(_OPENMP) && (!defined(TRISYCL_NO_BARRIER) && !defined(_MSC_VER))
```

Comment:
They might, but a few lines down is `#pragma omp task firstprivate(th_id)` and `task` is an OpenMP 3.0 construct, so no can do. If we find an OpenMP 2.0 implementation to launching work-groups in the correct way, then we can remove this extra define.

As a first approach, I did not want to enhance the lib this way. I usually do:

1. Make it work. (compile)
2. Make it right. (actually do what you intended)
3. Make it fast.

This change is just no. 1.

ps.: If we are to introduce OpenMP 2.0/3.0/4.0(+offload)/ParallelSTL/C++AMP? implementations, we should factor API-specific functions through some means before hell breaks loose. (Should be a different pull-request.)